src/openforms/emails/digest.py

```diff
@@ -389,16 +400,37 @@ def collect_invalid_logic_rules() -> list[InvalidLogicRule]:
 
         form_logics_vars = []
         for logic in form_logics:
-            form_logics_vars += introspect_json_logic(
-                logic.json_logic_trigger
-            ).get_input_keys()
+            logic_introspection_result = introspect_json_logic(logic.json_logic_trigger)
+
+            if isinstance(logic_introspection_result, str):
```

Comment:
so this approach still doesn't really work in terms of code elegance - you've now essentially traded a `try: except` with a `isinstance` check - and the handling code is still duplicated (reporting, logging...)